### PR TITLE
Azure pipeline CI for iMX 6,7, and 8 firmware

### PR DESCRIPTION
**Pull request by dmcilvaney**
First swing at Azure pipeline CI.

Still needs to be hooked into the ms-iot Azure devops services to run.

Requires pull request https://github.com/ms-iot/imx-iotcore/pull/60 to run correctly.

iMX8 is updated for most recent MU update (1903) whic is not available quite yet so iMX8 is expected to fail.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 Windows 10 IoT Core for NXP i.MX Processors
 ==============
+<PUT PIPELINE STATUS BADGE HERE!>
```
Comment (christopherco) [inline README.md:3]: Leave this part out. We'll add it once the pipeline is up and running.

**Review by christopherco — CHANGES_REQUESTED** [review]

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,209 @@
+trigger:
+- master
+
+jobs:
+  - job: Build8
+    displayName: Build iMX 8
```
Comment (neilsh-msft) [inline azure-pipelines.yml:6]: We're building the iMX 8M, the 8 is a different beast.

File: README.md (hunk 1 above)
Comment (dmcilvaney) [reply README.md:3]: Talked with Chris, we should be able to replace this with the badge as soon as the pull request is done. Leaving it in.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        export GCC5_AARCH64_PREFIX=$(Build.SourcesDirectory)/gcc-linaro-7.2.1-2017.11-x86_64_aarch64-linux-gnu/bin/aarch64-linux-gnu-
+        sudo python3.7 -m pip3 install -r requirements.txt --upgrade
+        python3.7 NXP/MCIMX8M_EVK_4GB/PlatformBuild.py --setup
+
+        cd MU_BASECORE
+        make -C BaseTools
+        cd ..
+
+        python3.7 NXP/MCIMX8M_EVK_4GB/PlatformBuild.py TOOL_CHAIN_TAG=GCC5 BUILDREPORTING=TRUE BUILDREPORT_TYPES="PCD" TARGET=RELEASE MAX_CONCURRENT_THREAD_NUMBER=20 BLD_*_CONFIG_NOT_SECURE_UEFI=1
+      displayName: Build UEFI
```
Comment (neilsh-msft) [inline azure-pipelines.yml:124]: Should there be a popd here?

File: azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+    - script: |
+        pushd ../mu_platform_nxp/Build/MCIMX8M_EVK_4GB/RELEASE_GCC5/FV
+        cp $(Build.SourcesDirectory)/build/firmware/its/uefi_imx8_unsigned.its .
+        $(Build.SourcesDirectory)/../u-boot/tools/mkimage -f uefi_imx8_unsigned.its -r uefi.fit
```
Comment (neilsh-msft) [inline azure-pipelines.yml:129]: Should there be a popd here?

**Review by neilsh-msft — APPROVED** [review]

File: azure-pipelines.yml (hunk 2 above)
Comment (dmcilvaney) [reply azure-pipelines.yml:6]: We can split out different builds using the matrix, the jobs will be of the format "Build iMX 8 <TARGET NAME>". In this case Build iMX 8 NXPEVK_iMX8M_4GB. I assume once we have iMX8 buttoned down a bit more we will shift it to a makefile which can handle the differences.

File: azure-pipelines.yml (hunk 3 above)
Comment (dmcilvaney) [reply azure-pipelines.yml:124]: If anything the other pushd/popds are not needed. Each script task starts in the root directory (imx-iotcore checkout directory).

**Pull request merged**